symphony/bdk/core/service/presence/presence_service.py

```diff
@@ -0,0 +1,176 @@
+        `Get User Presence <https://developers.symphony.com/restapi/reference#user-presence-v3`_
+        """
+        return await self._presence_api.v3_user_uid_presence_get(uid=user_id,
+                                                                 session_token=await self._auth_session.session_token,
+                                                                 local=local)
+
+    async def external_presence_interest(self, user_ids: List[int]):
+        """ Register interest in a list of external users to get their presence info.
+
+        :param user_ids: List of user ids to be registered.
+        `External Presence Interest <https://developers.symphony.com/restapi/reference#register-user-presence-interest`_
+        """
+        await self._presence_api.v1_user_presence_register_post(session_token=await self._auth_session.session_token,
+                                                                uid_list=user_ids)
+
+    async def set_presence(self, status: PresenceStatus, soft: bool):
+        """ Set the presence info of the calling user.
+
+        :param status: The new presence state for the user.
+        Possible values are AVAILABLE, BUSY, AWAY, ON_THE_PHONE, BE_RIGHT_BACK, IN_A_MEETING, OUT_OF_OFFICE, OFF_WORK.
+
+        :param soft: If true, the user's current status is taken into consideration. If the user is currently OFFLINE,
+        the user's presence will still be OFFLINE, but the new presence will take effect when the user comes online. If
+        the user is currently online, the user's activity state will be applied to the presence if applicable. (e.g. if
+        you are setting their presence to AVAILABLE, but the user is currently idle, their status will be represented as
+        AWAY)
+        :return: Presence info of the calling user.
+        `Set Presence <https://developers.symphony.com/restapi/reference#set-presence"`_
+        """
+        presence_status: V2PresenceStatus = V2PresenceStatus(category=status.value)
+        return await self._presence_api.v2_user_presence_post(session_token=await self._auth_session.session_token,
+                                                              presence=presence_status,
+                                                              soft=soft)
+
+    async def create_presence_feed(self):
+        """ Creates a new stream capturing online status changes ("presence feed") for the company (pod) and returns the
+        ID of the new feed. The feed will return the presence of users whose presence status has changed since it was
+        last read.
+
+        :return: Presence feed Id
+        `Create Presence Feed <https://developers.symphony.com/restapi/reference#create-presence-feed"`_
+        """
+        string_id = await self._presence_api.v1_presence_feed_create_post(
+            session_token=await self._auth_session.session_token)
+        return string_id.id
+
+    async def read_presence_feed(self, feed_id):
```

Comment:
can we add type hints to this method and the one after

Reply:
added arguments and returns type hints to all functions